### PR TITLE
Base layout + table + paging

### PR DESCRIPTION
**Pull request by Ivanekrasov**
ADDRESS | base_layout

### CODE REVIEW

File: src/js/components/Header.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+import IconButton from '@material-ui/core/IconButton';
+import MenuIcon from '@material-ui/icons/Menu';
+
+class Header extends React.Component {
+    render() {
+        return(
+            <AppBar position="sticky" className='table-header'>
+                <Toolbar>
```
Comment (AlexandraSusloparova) [inline src/js/components/Header.js]: ololo

File: src/js/components/App.js [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 
 class App extends Component {
   render() {
     return (
-      <div>
-        <h1>Hello world!</h1>
-        <button>123</button>
-      </div>
-    );
+        <Fragment>
```
Comment (Ivanekrasov) [inline src/js/components/App.js]: You can just use <> instead of <Fragment>

File: src/js/components/NasaTable.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        const from = rowsPerPage * page
+        const visibleRows = this.getDataToShow(data.table, from, rowsPerPage)
+        this.setState({ data, visibleRows })
+    }
+
+    render() {
+        const { data, page, rowsPerPage, visibleRows, open } = this.state
+        const { headers, table = [] } = data
+        return(
+            <React.Fragment>
```
Comment (Ivanekrasov) [inline src/js/components/NasaTable.js]: same

File: src/js/components/ImageDialog.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import React, { Component } from 'react'
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import Button from '@material-ui/core/Button';
+
+class ImageDialog extends Component {
+
```
Comment (Ivanekrasov) [inline src/js/components/ImageDialog.js:28]: extra empty line

File: src/js/components/NasaTable.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+class NasaTable extends Component {
+
+    constructor (props) {
+        super(props)
+        this.state = {
+            data: {},
+            visibleRows: null,
+            page: 0,
+            rowsPerPage: 10,
+            open: false,
```
Comment (Ivanekrasov) [inline src/js/components/NasaTable.js]: Call it more declarative, related to its state like isModalOpened or isModalVisible etc

File: src/js/components/NasaTable.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+            onLink: false
+        }
+    }
+
+    handleClose = () => this.setState({ open: false })
+
+    getDataToShow = (table, from, rowsRepPage) => table.filter((row, i) => i >= from && i < from + rowsRepPage)
+
+    nextPage = async () => {
```
Comment (Ivanekrasov) [inline src/js/components/NasaTable.js]: why do you need async here?

File: src/js/components/NasaTable.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        const { rowsPerPage, page, data } = this.state
+        const from = rowsPerPage * (page + 1)
+        const visibleRows = this.getDataToShow(data.table, from, rowsPerPage)
+        this.setState({ visibleRows, page: page + 1 })
+    }
+
+    prevPage = async () => {
```
Comment (Ivanekrasov) [inline src/js/components/NasaTable.js]: and here?

File: src/js/components/NasaTable.js [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+        const { rowsPerPage, page, data } = this.state
+        const from = rowsPerPage * (page - 1)
+        const visibleRows = this.getDataToShow(data.table, from, rowsPerPage)
+        this.setState({ visibleRows, page: page - 1 })
+    }
+
+    changeRow = async (e) => {
```
Comment (Ivanekrasov) [inline src/js/components/NasaTable.js]: and here?

File: src/js/components/NasaTable.js [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+        if (i !== 0) {
+            return (<TableCell key={i}>{ cell }</TableCell>)
+        }
+        return (
+            <TableCell
+                key={i}
+                onClick={this.openDialog}
+                onMouseEnter={ () => this.setState({onLink: true})}
+                onMouseLeave={ () => this.setState({onLink: false})}
+                className={`first-item ${this.state.onLink && 'on-link'}`}
```
Comment (Ivanekrasov) [inline src/js/components/NasaTable.js]: That's too complicated for this feature. Try to use hover pseudo class

File: src/js/index.js [hunk 10]
```diff
@@ -1,6 +1,5 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-
```
Comment (Ivanekrasov) [inline src/js/index.js:3]: Why did you change this?

File: src/scss/header.scss [hunk 11]
```diff
@@ -0,0 +1,10 @@
+.table-header.table-header {
+    background-color: #004d40;
+}
+
+.sasha {
```
Comment (Ivanekrasov) [inline src/scss/header.scss]: =)

File: src/js/components/NasaTable.js [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+                    rowsPerPage={rowsPerPage}
+                    page={page}
+                    onChangeRowsPerPage={this.changeRow}
+                    backIconButtonProps={{
+                        'aria-label': 'Previous Page',
+                        'onClick': this.prevPage
+                    }}
+                    nextIconButtonProps={{
+                        'aria-label': 'Next Page',
+                        'onClick': this.nextPage
```
Comment (Ivanekrasov) [inline src/js/components/NasaTable.js]: In JavaScript we have keys of string type by default. You don't need to wrap them with quotes

File: src/js/components/NasaTable.js [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+                        </TableRow>
+                    </TableHead>
+                    <TableBody>
+                        { visibleRows && visibleRows.map((row, i) => <TableRow key={i}>
+                            { row.info.map((cell, i) => this.fillCells(cell, i)) }
+                        </TableRow>) }
+                    </TableBody>
+                </Table>
+                <TablePagination
+                    rowsPerPageOptions={[5, 10, 25]}
```
Comment (Ivanekrasov) [inline src/js/components/NasaTable.js]: move it to a separate class property please

File: src/js/components/NasaTable.js [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+        const res = await getNasaData();
+        const data = await res.json();
+        const from = rowsPerPage * page
+        const visibleRows = this.getDataToShow(data.table, from, rowsPerPage)
+        this.setState({ data, visibleRows })
+    }
+
+    render() {
+        const { data, page, rowsPerPage, visibleRows, open } = this.state
+        const { headers, table = [] } = data
```
Comment (Ivanekrasov) [inline src/js/components/NasaTable.js]: You can do nested destructuring like
`{ data: { headers, table=[] }, ... `

File: src/js/components/TablePage.js [hunk 15]
```diff
@@ -0,0 +1,12 @@
...
+
+class TablePage extends React.Component {
+    render() {
+        return (
+            <NasaTable/>
+        )
+    }
+}
+
+export default TablePage;
```
Comment (Ivanekrasov) [inline src/js/components/TablePage.js]: Will you use some extra functionality in this component or it'll be just a component composition? 
In case it's used just to render components convert it into functional component

File: src/js/components/App.js [hunk 16]
```diff
@@ -1,14 +1,16 @@
-import React, { Component } from 'react';
+import React, { Component, Fragment } from 'react';
```
Comment (Ivanekrasov) [inline src/js/components/App.js]: add empty line after

File: src/js/components/App.js [hunk 17]
```diff
@@ -1,14 +1,16 @@
...
+import TablePage from './TablePage';
+import Header from './Header'
```
Comment (Ivanekrasov) [inline src/js/components/App.js]: and one more empty line after this

File: src/scss/table.scss [hunk 18]
```diff
@@ -0,0 +1,14 @@
+.table-headers.table-headers {
+  background-color: #26a69a;
+  color: white;
+}
+
+.first-item.first-item {
```
Comment (Ivanekrasov) [inline src/scss/table.scss:6]: Why do you have duplicate classes here and upper(table-headers)?

File: src/js/components/NasaTable.js [hunk 19]
```diff
@@ -0,0 +1,117 @@
+import React, { Component } from 'react';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import TablePagination from '@material-ui/core/TablePagination';
```
Comment (Ivanekrasov) [inline src/js/components/NasaTable.js:7]: add empty line after

File: src/js/components/NasaTable.js [hunk 20]
```diff
@@ -0,0 +1,117 @@
...
+import ImageDialog from './ImageDialog';
+import { getNasaData } from '../api/table'
```
Comment (Ivanekrasov) [inline src/js/components/NasaTable.js]: semicolon

File: src/js/components/Header.js [hunk 21]
```diff
@@ -0,0 +1,26 @@
...
+                        <MenuIcon />
+                    </IconButton>
+                    <Typography variant="h6">
+                        Space table
+                    </Typography>
+                </Toolbar>
+            </AppBar>
+        )
+    }
+}
```
Comment (Ivanekrasov) [inline src/js/components/Header.js]: Same question like for TablePage

File: src/js/components/NasaTable.js [hunk 22]
```diff
@@ -0,0 +1,117 @@
...
+    openDialog = () => this.setState({ open: true })
+
+    componentDidMount = async () => {
+        const { rowsPerPage, page } = this.state;
+        const res = await getNasaData();
+        const data = await res.json();
+        const from = rowsPerPage * page
+        const visibleRows = this.getDataToShow(data.table, from, rowsPerPage)
+        this.setState({ data, visibleRows })
+    }
```
Comment (Ivanekrasov) [inline src/js/components/NasaTable.js]: React lifecycle hooks should be ordinary functions, you can assign wrong context this way

**Review by Ivanekrasov — COMMENTED** [review]
Add semicolons to any arrow fuction you've declared

File: src/js/api/table.js [hunk 23]
```diff
@@ -0,0 +1,90 @@
+export const getNasaData = () => Promise.resolve({ json: () => Promise.resolve(nasaData)});
```
Comment (Fr0stmourne) [inline src/js/api/table.js]: please use export default syntax

File: src/js/components/Header.js [hunk 24]
```diff
@@ -0,0 +1,26 @@
...
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+import IconButton from '@material-ui/core/IconButton';
+import MenuIcon from '@material-ui/icons/Menu';
+
+class Header extends React.Component {
+    render() {
+        return(
+            <AppBar position="sticky" className='table-header'>
```
Comment (Fr0stmourne) [inline src/js/components/Header.js]: please avoid inline styles

File: src/js/components/ImageDialog.js [hunk 25]
```diff
@@ -0,0 +1,39 @@
...
+    render() {
+        const { open, onClose } = this.props
+        return(
+            <Dialog
+                open={open}
+                onClose={onClose}
+                aria-labelledby="alert-dialog-title"
+                aria-describedby="alert-dialog-description"
+            >
+                <DialogTitle id="alert-dialog-title">{"Mars"}</DialogTitle>
```
Comment (Fr0stmourne) [inline src/js/components/ImageDialog.js]: please use a variable or remove redundant brackets and quotes

File: config/webpack.base.conf.js [hunk 26]
```diff
@@ -29,7 +29,7 @@ module.exports = {
     rules: [
       {
         test: /\.js$/,
-        loader: ['babel-loader', 'eslint-loader'],
```
Comment (zanhesl) [inline config/webpack.base.conf.js]: please return lint to your branch

File: src/scss/main.scss [hunk 27]
```diff
@@ -2,4 +2,6 @@
 @import "utils/variables";
 @import "utils/mixins";
 @import "utils/fonts";
-@import "utils/base.scss";
\ No newline at end of file
+@import "utils/base.scss";
```
Comment (zanhesl) [inline src/scss/main.scss:5]: consider creating a folder for each component with index.js and {componentName}.scss

File: src/scss/header.scss [hunk 28]
```diff
@@ -0,0 +1,10 @@
...
+    height: 20px;
+    width: 20px;
+    background-color: red;
+    z-index: 1000;
```
Comment (zanhesl) [inline src/scss/header.scss]: do you really need this?

File: src/js/components/App.js (hunk 2 above)
Comment (AlexandraSusloparova) [reply src/js/components/App.js]: done

File: src/js/components/NasaTable.js (hunk 3 above)
Comment (AlexandraSusloparova) [reply src/js/components/NasaTable.js]: done

File: src/js/components/ImageDialog.js (hunk 4 above)
Comment (AlexandraSusloparova) [reply src/js/components/ImageDialog.js:28]: done

File: src/js/components/NasaTable.js (hunk 5 above)
Comment (AlexandraSusloparova) [reply src/js/components/NasaTable.js]: done

File: src/js/components/NasaTable.js (hunk 6 above)
Comment (AlexandraSusloparova) [reply src/js/components/NasaTable.js]: It expected that we will fetch data on page changing, forgot to remove, I have already fixed it.

File: src/js/components/NasaTable.js (hunk 7 above)
Comment (AlexandraSusloparova) [reply src/js/components/NasaTable.js]: explained in previous conversation

File: src/js/components/NasaTable.js (hunk 8 above)
Comment (AlexandraSusloparova) [reply src/js/components/NasaTable.js]: explained in previous conversation

File: src/js/index.js (hunk 10 above)
Comment (AlexandraSusloparova) [reply src/js/index.js:3]: It seems to me that it looks like better without empty line, but if it is important for you then i will return it.

File: src/scss/header.scss (hunk 11 above)
Comment (AlexandraSusloparova) [reply src/scss/header.scss]: ooops, sorry

File: src/js/components/NasaTable.js (hunk 12 above)
Comment (AlexandraSusloparova) [reply src/js/components/NasaTable.js]: Okay, it will work for onClick, but in case of aria-label we have dash and it is not valid syntax for object's keys.

File: src/js/components/NasaTable.js (hunk 13 above)
Comment (AlexandraSusloparova) [reply src/js/components/NasaTable.js]: Can you clarify please your request, as far as I understood you want me to move [ 5, 10, 25 ] to constructor either you want to put it like props, now I will put it to constructor like this.rowsValue = [ 5, 10, 25 ].

File: src/js/components/NasaTable.js (hunk 14 above)
Comment (AlexandraSusloparova) [reply src/js/components/NasaTable.js]: It seems to me that it doesn't make sense, can you explain the reason?

File: src/js/components/App.js (hunk 16 above)
Comment (AlexandraSusloparova) [reply src/js/components/App.js]: Do you want to separate custom components from libraries?

File: src/js/components/App.js (hunk 17 above)
Comment (AlexandraSusloparova) [reply src/js/components/App.js]: done

File: src/scss/table.scss (hunk 18 above)
Comment (AlexandraSusloparova) [reply src/scss/table.scss:6]: it some feature to get more priority for selector to override library's styles.

File: src/js/components/NasaTable.js (hunk 19 above)
Comment (AlexandraSusloparova) [reply src/js/components/NasaTable.js:7]: done

File: src/js/components/NasaTable.js (hunk 20 above)
Comment (AlexandraSusloparova) [reply src/js/components/NasaTable.js]: done

File: src/js/components/NasaTable.js (hunk 22 above)
Comment (AlexandraSusloparova) [reply src/js/components/NasaTable.js]: done

File: src/js/api/table.js (hunk 23 above)
Comment (AlexandraSusloparova) [reply src/js/api/table.js]: It will make sense if we will have more than 1 function here and it seems to me that we won't use any other functions here. But if you mind let me know and I will change it.

File: src/js/components/ImageDialog.js (hunk 25 above)
Comment (AlexandraSusloparova) [reply src/js/components/ImageDialog.js]: okay, to be honest this component need a big improvement, cause it is only prototype with  lot of stubs, and I will improve it right after some clarification about data format

File: src/scss/main.scss (hunk 27 above)
Comment (AlexandraSusloparova) [reply src/scss/main.scss:5]: done

File: src/scss/header.scss (hunk 28 above)
Comment (AlexandraSusloparova) [reply src/scss/header.scss]: it has fixed in previos conversation

File: src/js/components/Header.js (hunk 21 above)
Comment (AlexandraSusloparova) [reply src/js/components/Header.js]: resolved

File: src/js/api/table.js (hunk 23 above)
Comment (AlexandraSusloparova) [reply src/js/api/table.js]: resolved

File: config/webpack.base.conf.js (hunk 26 above)
Comment (AlexandraSusloparova) [reply config/webpack.base.conf.js]: resolved

File: src/js/components/TablePage.js (hunk 15 above)
Comment (AlexandraSusloparova) [reply src/js/components/TablePage.js]: resolved
